### PR TITLE
Add diagram for game loop

### PR DESCRIPTION
**Pull request by sharkwouter**
This adds a basic game loop diagram made in plantuml

### CODE REVIEW

File: doc/gameloop.uml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+@startuml
+title Game loop
+
+start
+
+:Generate board;
+:Remove matches from board;
+:Draw board;
```
Comment (protheory8) [inline doc/gameloop.uml]: It seems like board is being drawn only once, if I'm not wrong.

Comment (sharkwouter) [reply doc/gameloop.uml]: No, the board is of course drawn every frame, but I specified draw board at the top to clarify that the board shouldn't be drawn until it's ready. I'll change it to start drawing board then.

Comment (protheory8) [reply doc/gameloop.uml]: I think I just got confused, there is no arrow pointing to the start of the game loop.

**Review by protheory8 — APPROVED** [review]
I'm confused, but OK, it's fine.

**Pull request merged**